### ISSUE TITLE
Suggestion: Repentance iOS version?

### ISSUE THREAD

**Issue by PuchRM89** [wofsauge/External-Item-Descriptions#652]
Greetings. I noticed that among the instructions for applying the mod, the iOS version of the game is also present but it stops at Rebirth. Is it possible to update the mod so that it can also be used for Repentance iOS? A thousand thanks.

**Comment by wofsauge** [comment]
The iOS version of the game does not have a modding API, therefore a Repentance version of the mod would require us to use the old rebirth method of modding. that is done by replacing each item sprite with a much bigger version, that also adds a part that contains the item descriptions as a text box like shown here:
![collectibles_121_oddmushroomlarge](https://github.com/wofsauge/External-Item-Descriptions/assets/245725/5ec8ff1a-6d45-469f-ba1e-33b93cbff828)

The problem is, that back then, those images where created and adjusted manually, to optimize every pixel of screen space, because they are always visible and therefore are very easy to overlap each other.

for a repentance version of that, we would need to create all 700+ collectible and 200+ trinket descriptions from scratch, because a lot of item effects changed over the years. that task is extremely time consuming, if its not automated by a script of some sort.

I currently dont have the time to invest in making such a script, therefore i will not create a Repentance version for iOS in the forseeable future.

If someone else is willing to tackle this task, they can of course do so

**Comment by srose69** [comment]
Any sources?)

**Comment by wofsauge** [comment]
as described aboth, the iOS version of the mod would need to use the old way of displaying descriptions. basically by putting them into the item sprite directly in a neat format.

i guess the "easiest" way to achieve that would be to write a script that reads the current EID descriptions, and generates the altered item sprites with the description added to them.

I dont know how big the userbase for iOS Isaac even is. especially those, who are able to mod the game (requires jailbreak as far as i know). The userbase therefore is fairly small

**Comment by srose69** [comment]
If there were sources, it would probably be possible to try to rewrite them a bit. Or just port the Isaac PC to iOS, with all the goodies.

**Comment by emp0ry** [comment]
@wofsauge I created native External Item Descriptions for The Binding of Isaac: Repentance on iOS
https://github.com/emp0ry/IsaacExternalItemDescriptionsiOS/releases/tag/v0.1.0

**Comment by wofsauge** [comment]
Great job! Thank you very much for creating this :)
I've added a link to the repo to the official installation guide of EID :)

**Comment by emp0ry** [comment]
@wofsauge thx! can I add to description files grom your repo to my release?

**Comment by wofsauge** [comment]
Yes you can add them. Just keep in mind to use a version of the description files that is compatible with the ios version of the game. I dont know if the steam rep version is on the same version number as the ios version of the game.

**Comment by emp0ry** [comment]
I think it’s more or less the same version as Steam, and the save files are compatible with the Steam version. I’ve also made Steam Cloud sync work on iOS 😁 It’s on my GitHub.

**Comment by wofsauge** [comment]
Very cool :) maybe you can add a screenshot/video of the mod in action on ios as well in here, and maybe advertise this new mod on reddit and discord?

**Comment by emp0ry** [comment]
https://github.com/user-attachments/assets/f85426a3-4bdd-40d3-9a20-1d9fd1b0ee1c
Yes, can post it on Reddit and Discord. This is first release version and hasn't been tested much.

**Comment by wofsauge** [comment]
Looks very good and useful so far. Good job 👍

**Comment by emp0ry** [comment]
Thanks!!!

**Comment by emp0ry** [comment]
https://www.reddit.com/r/bindingofisaac/s/8dDBPLukcj
reddit post

**Comment by emp0ry** [comment]
<img width="1434" height="660" alt="Image" src="https://github.com/user-attachments/assets/47472c5f-1700-417e-8fff-ed101d6b9d39" />

**Comment by wofsauge** [comment]
Very awesome progress! Good job

**Comment by emp0ry** [comment]
Thanks! Glad you like it :)
